### DISCUSSION TITLE
Having difficulty mocking internal modules within my project

### DISCUSSION THREAD

**Discussion by AlexMachin1997** [vitest-dev/vitest#2986] [Q&A]
Hi everyone,

So I'm in the process of porting some tests from Jest to Vitest and I'm having some difficulties mocking internal files within my project.

I've got a component which consumes a configuration files, within the test I'm trying to mock the configuration files. However, when I run the tests the mocked values appear blank when used within the component.

**How I'm mocking the file:**

```js
// TopMenu.test.jsx

vi.mock("../../utils/Configuration.js", async () => {
  const actual = await vi.importActual("../../utils/Configuration.js");

  return {
    ...actual,
    externalUris: {
      ...actual.externalUris,
      investigatorPortalUri: "https://investigatorportal.com/",
      userManagementUri: "https://usermanagementuri.com/",
    },
  };
});

```
I'm not sure where I'm going wrong, vi.mock() and in that function I'm porting the original file and then appending my hardcoded values in the return statement.

Any comments in reguards to my issue would be greatly appreciated, I've been stuck with this for a while now and I've not been able to make any progress.

**Comment by sheremet-va** [comment]
How do you import the configuration?

```ts
import configuration from './utils/Configuration.js'
// or
import * as configuration from './utils/Configuration.js'
// or
import { externalUris } from './utils/Configuration.js'
```

**Reply by AlexMachin1997** [reply]
Yeah I have the first.

This is how it's imported within the component: 
`import Configuration from "../../utils/Configuration";`

**Reply by sheremet-va** [reply]
Then you should export your values on `default` property. See example: https://vitest.dev/api/vi.html#vi-mock

**Reply by AlexMachin1997** [reply]
Wow seriously...

How on earth did I miss this key part of the docs, thank you so much @sheremet-va 

After all that confusion it's just boiled down to the `default` property being missed (Face palms self). Jest much ave handled this internally then since you didn't need to do this for default's in Jest.

```
vi.mock("../../utils/Configuration.js", async () => {
  const actual = await vi.importActual("../../utils/Configuration.js");

  return {
    default: {
      ...actual,
      externalUris: {
        ...actual.externalUris,
        investigatorPortalUri: "https://investigatorPortal.com",
        userManagementUri: "https://userManagementUri.com",
      },
    },
  };
});

```

**Reply by sheremet-va** [reply]
Jest uses CommonJS, where default export is the same as the module, which is not true for ESM. Your example would look more like this:

```
    ...actual,
    default: {
      ...actual.default,
      externalUris: {
        ...actual.default.externalUris,
        investigatorPortalUri: "https://investigatorPortal.com",
        userManagementUri: "https://userManagementUri.com",
      },
    },
```
